app/models/node.rb

```diff
@@ -123,8 +123,13 @@ def slug_from_path
     path.split('/').last
   end
 
-  def has_a_tag(name)
-    return tags.where(name: name).size.positive?
+  def self.hidden_response_node_ids
+    Node.joins(:node_tag)
+        .joins('LEFT OUTER JOIN term_data ON term_data.tid = community_tags.tid')
+        .select('node.nid')
```

Comment:
Looks great! However i think this may need more selected fields -- https://guides.rubyonrails.org/active_record_querying.html#selecting-specific-fields says it actually changes the `SELECT X` section of the SQL query, so we may need all those used in the joins and where fields.

I read your response about this somewhere, i think... but can't find it now? Maybe you deleted it, or was it in Gitter? gosh, my brain isn't working so clearly right now as I'm deep in this Travis thing... sorry!

Just want to confirm this here though after checking the Rails docs!

Reply:
It was in my other [PR](https://github.com/publiclab/plots2/pull/8848#discussion_r543489655) 😄 

Thanks, I'll have a look at the doc...and update the PR.